velox/exec/PartitionedOutputBufferManager.cpp

```diff
@@ -544,7 +554,14 @@ void PartitionedOutputBufferManager::getData(
     uint64_t maxBytes,
     int64_t sequence,
     DataAvailableCallback notify) {
-  getBuffer(taskId)->getData(destination, maxBytes, sequence, notify);
+  auto buffer = getBufferIfExists(taskId);
+  if (buffer) {
+    buffer->getData(destination, maxBytes, sequence, notify);
+  }
+  std::vector<std::unique_ptr<folly::IOBuf>> pages;
+  auto param = std::make_unique<DataAvailableCallbackParam>(
+      std::move(pages), sequence, ERR_BUFFER_NOT_FOUND);
```

Comment:
It would be simpler to just return 'false' to indicate that buffers for the task do not exist and therefore the 'notify' callback cannot be queued. See my other comment.